### ISSUE TITLE
bug: How to enable tracing for a container created with openllm?

### ISSUE THREAD

**Issue by martinmr** [bentoml/OpenLLM#447]
I am looking to enable tracing to a service that deploys a container built with openllm?

So I added this config to my service (the join function is just turning all these arguments into the proper format):

```
BENTOML_CONFIG_OPTIONS = join(" ", [
      "api_server.workers=8",
      "api_server.backlog=2048",
      "api_server.traffic.timeout=60",
      "api_server.traffic.max_concurrency=100",
      "api_server.logging.access.enabled=true",
      "runners.batching.enabled=true",
      "runners.batching.max_batch_size=100",
      "runners.batching.max_latency_ms=500",
      "tracing.exporter_type=otlp",
      "tracing.sample_rate=1.0",
      "tracing.otlp.protocol=grpc",
      "tracing.otlp.endpoint=http://$DD_AGENT_HOST:4317",
      "tracing.otlp.grpc.insecure=true",
      "tracing.otlp.grpc.headers[0][0]=grpc-encoding",
      "tracing.otlp.grpc.headers[0][1]=gzip",
    ])
```

When I deploy my changes, I get this error:

```
aceback (most recent call last):                                                                                                           │
│   File "/opt/conda/lib/python3.9/runpy.py", line 197, in _run_module_as_main                                                                 │
│     return _run_code(code, main_globals, None,                                                                                               │
│   File "/opt/conda/lib/python3.9/runpy.py", line 87, in _run_code                                                                            │
│     exec(code, run_globals)                                                                                                                  │
│   File "/opt/conda/lib/python3.9/site-packages/bentoml_cli/worker/http_api_server.py", line 197, in <module>                                 │
│     main()  # pylint: disable=no-value-for-parameter                                                                                         │
│   File "/opt/conda/lib/python3.9/site-packages/click/core.py", line 1157, in __call__                                                        │
│     return self.main(*args, **kwargs)                                                                                                        │
│   File "/opt/conda/lib/python3.9/site-packages/click/core.py", line 1078, in main                                                            │
│     rv = self.invoke(ctx)                                                                                                                    │
│   File "/opt/conda/lib/python3.9/site-packages/click/core.py", line 1434, in invoke                                                          │
│     return ctx.invoke(self.callback, **ctx.params)                                                                                           │
│   File "/opt/conda/lib/python3.9/site-packages/click/core.py", line 783, in invoke                                                           │
│     return __callback(*args, **kwargs)                                                                                                       │
│   File "/opt/conda/lib/python3.9/site-packages/bentoml_cli/worker/http_api_server.py", line 192, in main                                     │
│     config = uvicorn.Config(svc.asgi_app, **uvicorn_options)                                                                                 │
│   File "/opt/conda/lib/python3.9/site-packages/bentoml/_internal/service/service.py", line 421, in asgi_app                                  │
│     return HTTPAppFactory(self)()                                                                                                            │
│   File "/opt/conda/lib/python3.9/site-packages/bentoml/_internal/server/http_app.py", line 305, in __call__                                  │
│     app = super().__call__()                                                                                                                 │
│   File "/opt/conda/lib/python3.9/site-packages/bentoml/_internal/server/base_app.py", line 86, in __call__                                   │
│     middleware=self.middlewares,                                                                                                             │
│   File "/opt/conda/lib/python3.9/site-packages/bentoml/_internal/server/http_app.py", line 241, in middlewares                               │
│     tracer_provider=BentoMLContainer.tracer_provider.get(),                                                                                  │
│   File "/opt/conda/lib/python3.9/site-packages/simple_di/__init__.py", line 72, in get                                                       │
│     return self._provide()                                                                                                                   │
│   File "/opt/conda/lib/python3.9/site-packages/simple_di/providers.py", line 124, in _provide                                                │
│     value = super()._provide()                                                                                                               │
│   File "/opt/conda/lib/python3.9/site-packages/simple_di/providers.py", line 103, in _provide                                                │
│     return inject(self._func)(                                                                                                               │
│   File "/opt/conda/lib/python3.9/site-packages/simple_di/__init__.py", line 139, in _                                                        │
│     return func(*_inject_args(bind.args), **_inject_kwargs(bind.kwargs))                                                                     │
│   File "/opt/conda/lib/python3.9/site-packages/bentoml/_internal/configuration/containers.py", line 433, in tracer_provider                  │
│     from opentelemetry.exporter.otlp.proto.grpc import trace_exporter                                                                        │
│ ModuleNotFoundError: No module named 'opentelemetry.exporter'
```

This looks like a module not found error that's triggered when enabling tracing, and not something wrong with the config I passed.

**Comment by aarnphm** [comment]
See `bentoml containerize -h` for more information. If you wish to use tracing you need to do `bentoml containerize <bento> --enable-features tracing` so that the container will include tracing exporter dependencies.

**Comment by aarnphm** [comment]
we will need to add option for extra bentoml dependency here.

**Comment by aarnphm** [comment]
This has been fixed on main and will be included in 0.4.1
